Решение Фибоначчи

dimaymik opened the pull request:
Беккер Дмитрий

dimaymik committed:
Решение Фибоначчи

dimaymik committed:
Merge branch 'master' of https://github.com/dimaymik/scala-school

sergeypopov83 commented on src/main/scala/lectures/functions/Fibonacci.scala:
```diff
@@ -12,13 +14,13 @@ package lectures.functions
 
   // Task 2
-  def fibs(num: Int) = {
+  def fibs(num: Int): Int = {
```
Ок. 
Для общего развития посмотри, на типы BigInt, BigDecimal. В числовых задачах, неограниченных  сверху по значению, бывает полезно использовать их

dimaymik committed:
Carried

dimaymik committed:
Carried

dimaymik committed:
Carried

dimaymik committed:
test

dimaymik committed:
Жесть

sergeypopov83 commented on src/main/scala/lectures/functions/Computations.scala:
```diff
@@ -41,12 +41,12 @@ object Computation extends App with Data {
 
-  def curriedComputation(filterData: String)(dataProducer: Array[String]): Array[String] = ???
+  def curriedComputation(filterData: String)(dataProducer: Array[String]): Array[String] = dataProducer.filter(dataitem =>
+    filterData.split(" ").contains(dataitem))
```
Верни, пожалуйста Thread.sleep во все имплементации. Они нам понадобятся

sergeypopov83 commented on src/main/scala/lectures/functions/Computations.scala:
```diff
@@ -55,11 +55,17 @@ object CurriedComputation extends App with Data {
-  def functionalComputation(filterData: String): (Array[String]) => Array[String] = ???
+  def functionalComputation(filterData: String): (Array[String]) => Array[String] = {
+    val splitFilterData = filterData.split(" ")
+    def funcComp(dataProducer: Array[String]): Array[String] = {
```
лучше val вместо def. Я про это не рассказывал, но на def больше накладных  расходов.

sergeypopov83 commented on src/main/scala/lectures/functions/SQLAPI.scala:
```diff
@@ -37,12 +37,22 @@ class SQLAPI(resource :String) {
+    logParameter(connection(resource).open().execute(sql))
+    //Java рулит
+    */
+    logParameter(resource)
```
попробуй скомпозировать так, что бы не было отдельного logParameter(resource)

dimaymik commented:
А можно я не буду лог, потому что пробовал и мозг начал плавиться)

> 13 марта 2017 г., в 15:44, Sergey Popov <notifications@github.com> написал(а):
> 
> @sergeypopov83 commented on this pull request.
> 
> In src/main/scala/lectures/functions/Computations.scala:
> 
> > @@ -41,12 +41,12 @@ object Computation extends App with Data {
>    */
>  object CurriedComputation extends App with Data {
>  
> -  def curriedComputation(filterData: String)(dataProducer: Array[String]): Array[String] = ???
> +  def curriedComputation(filterData: String)(dataProducer: Array[String]): Array[String] = dataProducer.filter(dataitem =>
> +    filterData.split(" ").contains(dataitem))
> Верни, пожалуйста Thread.sleep во все имплементации. Они нам понадобятся
> 
> In src/main/scala/lectures/functions/Computations.scala:
> 
> > @@ -55,11 +55,17 @@ object CurriedComputation extends App with Data {
>    */
>  object FunctionalComputation extends App with Data {
>  
> -  def functionalComputation(filterData: String): (Array[String]) => Array[String] = ???
> +  def functionalComputation(filterData: String): (Array[String]) => Array[String] = {
> +    val splitFilterData = filterData.split(" ")
> +    def funcComp(dataProducer: Array[String]): Array[String] = {
> лучше val вместо def. Я про это не рассказывал, но на def больше накладных расходов.
> 
> In src/main/scala/lectures/functions/SQLAPI.scala:
> 
> >  
>    val connection = (resource: String) => Connection(resource)
>  
> -  def execute(sql: String): String = ??? // use resource from constructor
> -
> +  def execute(sql: String): String = {
> +/*    logParameter(resource)
> +    logParameter(sql)
> +    logParameter(connection(resource).open().execute(sql))
> +    //Java рулит
> +    */
> +    logParameter(resource)
> попробуй скомпозировать так, что бы не было отдельного logParameter(resource)
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

sergeypopov83 commented:
Привет) Мозг и должен плавиться, это нормально. В заданиях нет ничего обязательного , это же  не школа). Просто ты не поймешь важную вещь, которую  поймут другие и у них будет преимущество.

dimaymik committed:
Stuf

dimaymik committed:
Evaluate.
